File: 1823-find-the-winner-of-the-circular-game/main.py
class Solution:
    def findTheWinner(self, n: int, k: int) -> int:
        
        friends = [i for i in range(1, n + 1)]
        idx = 0

        while len(friends) > 1:
            idx = (idx + k - 1) % len(friends) 
            friends.pop(idx)

        return friends[0]

File: 1823-find-the-winner-of-the-circular-game/main.go
func findTheWinner(n int, k int) int {
    friends := make([]int, n)
    for i := range n {
        friends[i] = i + 1
    }

    idx := 0

    for len(friends) > 1 {
        idx = (idx + k - 1) % len(friends)
        friends = append(friends[:idx], friends[idx+1:]...)
    }

    return friends[0]
}